slack-api-client/src/test/java/test_locally/api/audit/ApiTest.java

```diff
@@ -159,4 +166,39 @@ public void getActions() throws Exception {
         assertThat(stats.getAllCompletedCalls().get("actions"), is(1L));
     }
 
+    @Test
+    public void customizeExecutorService() throws Exception {
+        final AtomicBoolean called = new AtomicBoolean(false);
+        SlackConfig config = new SlackConfig();
+        config.setMethodsEndpointUrlPrefix("http://127.0.0.1:" + port + "/api/");
+        config.setAuditEndpointUrlPrefix("http://127.0.0.1:" + port + "/api/");
+        config.setExecutorServiceProvider(new ExecutorServiceProvider() {
+            @Override
+            public ExecutorService createThreadPoolExecutor(String threadGroupName, int poolSize) {
+                called.set(true);
+                return DaemonThreadExecutorServiceProvider.getInstance()
+                        .createThreadPoolExecutor(threadGroupName, poolSize);
+            }
+
+            @Override
+            public ScheduledExecutorService createThreadScheduledExecutor(String threadGroupName) {
+                return DaemonThreadExecutorServiceProvider.getInstance()
+                        .createThreadScheduledExecutor(threadGroupName);
+            }
+        });
+        Slack slack = Slack.getInstance(config);
+        {
+            ActionsResponse response = slack.audit(ValidToken).getActions();
+            assertThat(response.getError(), is(""));
+            // the sync client does not use ExecutorService under the hood
+            assertThat(called.get(), is(false));
```

Comment:
TODO: add assertions for executor service creation for metric database under the hood